_pytest/runner.py

```diff
@@ -131,9 +130,8 @@ def _update_current_test_var(item, when):
     var_name = 'PYTEST_CURRENT_TEST'
     if when:
         value = '{0} ({1})'.format(item.nodeid, when)
-        if _PY2:
-            # python 2 doesn't like null bytes on environment variables (see #2644)
-            value = value.replace('\x00', '(null)')
+        # don't allow null bytes on environment variables (see #2644, #2957)
+        value = value.replace('\x00', '(null)')
```

Comment:
i came to the conclusion thati this place is the **completely** wrong place to fix the issue

the real issue is that we allow bad test ids in the firs place - that is an full stop no go thing to begin with

Reply:
If we consider that null bytes make part of a "bad test ids" (which I think is reasonable) I agree this is not the bast place to put it. 

It is of course simple enough to refactor this later without much fuss. Would mind opening an issue @RonnyPfannschmidt?